tests/unit/base/DynamicObjectArrayNoTempl_unittest.cc

```diff
@@ -0,0 +1,34 @@
+/* This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Written (W) 2017 Leon Kuchenbecker
+ */
+
+#include <shogun/lib/DynamicObjectArray.h>
+#include <shogun/features/Subset.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+TEST(DynamicObjectArray,clone)
+{
+	// Base array
```

Comment:
superflous comment, just call it "base_array" (the name do_array doesnt help a lot)